File: habit-builder/app/Models/ProcrastinationAlert.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class ProcrastinationAlert extends Model
{
    use HasFactory;

    protected $fillable = [
        'user_id',
        'habit_id',
        'days_delayed',
        'severity',
        'message',
        'is_read',
        'sent_at',
    ];

    protected $casts = [
        'is_read' => 'boolean',
        'sent_at' => 'datetime',
    ];

    public function user(): BelongsTo
    {
        return $this->belongsTo(User::class);
    }

    public function habit(): BelongsTo
    {
        return $this->belongsTo(Habit::class);
    }

    public static function generateMessage(int $daysDelayed, string $habitName): string
    {
        if ($daysDelayed === 1) {
            return "🤔 Hôm nay bạn chưa hoàn thành '{$habitName}'. Có lẽ bạn đang trì hoãn?";
        } elseif ($daysDelayed === 2) {
            return "⚠️ Đã 2 ngày rồi! Bạn đang trì hoãn '{$habitName}' rõ ràng. Hãy bắt đầu ngay bây giờ!";
        } elseif ($daysDelayed === 3) {
            return "🚨 3 ngày không làm '{$habitName}'! Đây là dấu hiệu nghiêm trọng của thói quen trì hoãn.";
        } else {
            return "🔥 {$daysDelayed} ngày trôi qua mà chưa làm '{$habitName}'! Trì hoãn đang kiểm soát bạn. Hành động NGAY!";
        }
    }

    public static function getSeverity(int $daysDelayed): string
    {
        if ($daysDelayed >= 3) {
            return 'critical';
        } elseif ($daysDelayed >= 2) {
            return 'warning';
        }
        return 'info';
    }
}

File: home_habit_builder/lib/models/procrastination_alert.dart
class ProcrastinationAlert {
  final int id;
  final int userId;
  final int habitId;
  final int daysDelayed;
  final String severity;
  final String message;
  final bool isRead;
  final DateTime sentAt;
  final String? habitName;

  ProcrastinationAlert({
    required this.id,
    required this.userId,
    required this.habitId,
    required this.daysDelayed,
    required this.severity,
    required this.message,
    required this.isRead,
    required this.sentAt,
    this.habitName,
  });

  factory ProcrastinationAlert.fromJson(Map<String, dynamic> json) {
    return ProcrastinationAlert(
      id: json['id'],
      userId: json['user_id'],
      habitId: json['habit_id'],
      daysDelayed: json['days_delayed'],
      severity: json['severity'],
      message: json['message'],
      isRead: json['is_read'] ?? false,
      sentAt: DateTime.parse(json['sent_at']),
      habitName: json['habit']?['ten_thoi_quen'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'user_id': userId,
      'habit_id': habitId,
      'days_delayed': daysDelayed,
      'severity': severity,
      'message': message,
      'is_read': isRead,
      'sent_at': sentAt.toIso8601String(),
    };
  }

  // Helper methods
  bool get isCritical => severity == 'critical';
  bool get isWarning => severity == 'warning';
  bool get isInfo => severity == 'info';

  String get severityIcon {
    switch (severity) {
      case 'critical':
        return '🔥';
      case 'warning':
        return '⚠️';
      case 'info':
        return '💡';
      default:
        return '📌';
    }
  }
}
